src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIMeleeGuard.java

```diff
@@ -130,38 +133,54 @@ protected AIState huntDown()
         if (!targetEntity.isEntityAlive() || checkForWeapon())
         {
             targetEntity = null;
+            worker.setAIMoveSpeed((float) 1.0D);
+            return AIState.GUARD_GATHERING;
         }
 
-        if (targetEntity != null)
+        AbstractBuilding building = getOwnBuilding();
+        if(building != null && building instanceof BuildingGuardTower)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) No need to check for null before an instanceof [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ASimplifyConditional)